org.mixedrealitytoolkit.input/package.json

```diff
@@ -16,7 +16,7 @@
   "unity": "2021.3",
   "dependencies": {
     "com.microsoft.mrtk.graphicstools.unity": "0.5.12",
-    "org.mixedrealitytoolkit.core": "3.0.0",
+    "org.mixedrealitytoolkit.core": "3.0.1",
```

Comment:
Why force an upgrade to a patch version?  Scott and I have been wanted to only require updates if the package requires a feature change, or if there is a breaking change to a dependency.   Otherwise, we should give apps the option of when to upgrade.

Reply:
There's a new `internal` field in Core that Input relies on:

![image](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/3580640/519f93a2-2c1a-4429-9a45-5f62edf97253)

Because there's no change to the public API surface nor substantial new private functionality in Core, it doesn't match up with a minor version bump. Therefore, the dependency was increased to this patch version.

Reply:
Ack...I'm even more confused.

1. We already started the release for 3.0.1.   So you should update to MRTK Core to 3.0.2
2. How can outside assembly reference an internal field from a different assembly.  What's the point of marking something internal then?   I see that it's working, but I don't understand.

Reply:
1. Can do! (and done!)
2. We use the `InternalsVisibleTo` tag in an AssemblyInfo.cs file. We've mainly used this pattern to define something in Core that we want to be well-defined (or want to reserve the right to change the implementation of) across all packages but not expose publicly for one reason or another. From a quick check, it looks like the two classes that are accessed in Input (from Core) this way are HandsUtils.cs and PoseUtilities.cs.

    https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/blob/e2c3e55981876487e5e57c4f495f5aedcf8b52bf/org.mixedrealitytoolkit.core/AssemblyInfo.cs#L10

Reply:
Ahh...InternalsVisibleTo  ... You learn something new everyday.  Thanks for the explanation.

Reply:
Happy to help!